### PR TITLE
Replace individual setting of HELICS_VERSION_* variables with CMake project VERSION argument

### PR DESCRIPTION
**Pull request by nightlark**
This uses the project command's VERSION argument in CMake to set the version and component variables instead of setting every component individually. The VERSION argument can take a 4th component that is stored in HELICS_VERSION_TWEAK (and PROJECT_VERSION_TWEAK) variables.

### CODE REVIEW

**Comment by kdheepak** [comment]
Did you mean to merge this into `master`? Shouldn't changes go to `develop` first?

**Comment by nightlark** [comment]
Oh yes, that was meant to be develop.

File: CMakeLists.txt [hunk 1]
```diff
@@ -5,23 +5,20 @@
...
 set (HELICS_VERSION_BUILD )
 set (HELICS_DATE "06-30-18")
 
-if (NOT HELICS_VERSION_BUILD)
-	set (HELICS_VERSION "${HELICS_VERSION_MAJOR}.${HELICS_VERSION_MINOR}.${HELICS_VERSION_PATCH}")
-	set (HELICS_VERSION_UNDERSCORE "${HELICS_VERSION_MAJOR}_${HELICS_VERSION_MINOR}_${HELICS_VERSION_PATCH}")
-else()
-	set (HELICS_VERSION "${HELICS_VERSION_MAJOR}.${HELICS_VERSION_MINOR}.${HELICS_VERSION_PATCH}-${HELICS_VERSION_BUILD}")
-	set (HELICS_VERSION_UNDERSCORE "${HELICS_VERSION_MAJOR}_${HELICS_VERSION_MINOR}_${HELICS_VERSION_PATCH}-${HELICS_VERSION_BUILD}")
+set (HELICS_VERSION_UNDERSCORE "${HELICS_VERSION_MAJOR}_${HELICS_VERSION_MINOR}_${HELICS_VERSION_PATCH}")
```
Comment (phlptp) [inline CMakeLists.txt]: Can these be non-numeric?  specifically can HELICS_VERSION_TWEAK be non-numeric?

Comment (nightlark) [reply CMakeLists.txt]: Unfortunately no, it is all numeric.

**Pull request merged**